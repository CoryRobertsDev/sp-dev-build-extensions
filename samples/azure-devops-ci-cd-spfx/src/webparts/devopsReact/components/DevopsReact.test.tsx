import React from "react";
import 'jest';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import DevopsReact from './DevopsReact';
import { IDevopsReactProps } from "./IDevopsReactProps";
import { IUser } from "../services/IUser";

describe('component renders properly', () => {
  const addClickCallback = jest.fn();
  const props = {
    user: {
      FirstName: 'Jane',
      LastName: 'Smith',
      Email: 'Jane.Smith@contoso.com',
      Photo: '',
      JobTitle: 'General Manager',
    } as IUser,
    addNumbers: addClickCallback as any,
    mathValue: 10,
  } as IDevopsReactProps;
  const wrapper = shallow(<DevopsReact {...props} />);
  wrapper.find('button').at(0).simulate('click');
  test('add button is clicked', () => {
    expect(addClickCallback).toHaveBeenCalledTimes(1);
  });
  test('snapshot should match', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
