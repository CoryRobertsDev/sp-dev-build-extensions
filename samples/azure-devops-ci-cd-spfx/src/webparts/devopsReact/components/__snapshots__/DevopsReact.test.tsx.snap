// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders properly snapshot should match 1`] = `
<div
  className="devopsReact"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <span
          className="title"
        >
          Current complex value
        </span>
        <p
          className="subTitle"
        />
        <CustomizedDefaultButton
          onClick={[Function]}
          text="Add one to value"
        />
        <button
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <StyledPersonaBase
    hidePersonaDetails={false}
    imageUrl="data:image/jpg;base64,"
    presence={5}
    secondaryText="General Manager"
    size={15}
    tertiaryText="Jane.Smith@contoso.com"
    text="Jane Smith"
  />
</div>
`;
